### PR TITLE
fix: [CDS-75181]: Revisit implementation for removing comments form yaml

### PR DESCRIPTION
**Pull request by abosii-harness**
## Describe your changes
Replacing the logic from fully removing comments through serialization/deserialization by replacing comment content with Harness provided value. The main reason behind this feature was not to actually remove comments but to just not render the expressions in comments, so instead of completely removing this will achieve the same goal. 

While completely removing the comment is still achievable and doesn't require other changes than just removing YamlUtility#removeComments `46-48` we're still not sure if there is an use case where removing the comment without removing the new line may break the yaml itself (in theory it shouldn't but just want to on safe side here). Removing new line is a challenge as there are multiple use cases when it should be removed and when not.

### Tested cases
- [x] Customer use cases that reported some issues with this feature enabled
- [x] Customer use case who requested this feature
- [x] Multiple variations of the yaml (reference:  YamlUtilityTest

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51059)

### CODE REVIEW

**Comment by abosii-harness** [comment]
trigger checkstylecodeformat, commonchecks

**Comment by abosii-harness** [comment]
trigger uts

File: 125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sFilesCommentsHandler.java [hunk 1]
```diff
@@ -43,21 +39,13 @@ else if (ManifestType.Kustomize.equals(manifestType)) {}
...
-    Yaml yaml = new Yaml(options);
-    Map<String, String> map;
-    String str;
-    for (String values : valuesFiles) {
-      if (isNotEmpty(values)) {
-        map = yaml.load(values);
-        str = isNotEmpty(map) ? yaml.dump(map) : "";
-        modifiedValuesFiles.add(str);
-      }
+    List<String> modified = new ArrayList<>();
```
Comment (acasian-harness) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sFilesCommentsHandler.java:44]: nit: `ArrayList<>(valuesFiles.size())`

**Review by acasian-harness — APPROVED** [review]
@alexei-harness very clean and deep set of UT's. Great work!

File: 125-cd-nextgen/src/main/java/io/harness/cdng/k8s/yaml/YamlUtility.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  public String removeComments(String file) {
+    StringWriter outputFile = new StringWriter();
+    // snake yaml is used here just to speed implementation, but ideally this dependency should be removed
+    StreamReader streamReader = new StreamReader(new StringReader(file));
+    int prevToken = 0;
+    int scalarToken = 0;
+    boolean parsingQuotedScalar = false;
+    while (streamReader.peek() != 0) {
+      int token = streamReader.peek();
+      // if current token is '#' and we're there is no loose " or ' then everything els in line is a comment
```
Comment (GoelPratyush99) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/k8s/yaml/YamlUtility.java:40]: Nit: els -> else
 // if current token is '#' and we're there is no loose " or ' then everything else in line is a comment

**Review by GoelPratyush99 — APPROVED** [review]

**Comment by abosii-harness** [comment]
trigger checkstylecodeformat, commonchecks, uts

**Review by GoelPratyush99 — DISMISSED** [review]

**Review by acasian-harness — DISMISSED** [review]

**Comment by abosii-harness** [comment]
trigger checkstylecodeformat, commonchecks, uts

**Review by alebuh — DISMISSED** [review]
> Reviewed 3 of 6 files at r1, 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51059)* status: Some of the checks are too old. 8 file(s) to review. 1 of 2 approvals obtained. Waiting on team(s): k8s-team.

**Review by GoelPratyush99 — APPROVED** [review]

**Review by markoharness — DISMISSED** [review]

**Comment by abosii-harness** [comment]
trigger checkstylecodeformat, commonchecks, uts

**Review by alebuh — APPROVED** [review]

**Review by markoharness — APPROVED** [review]

**Review by abhinavsingh1196 — APPROVED** [review]

**Review by GoelPratyush99 — APPROVED** [review]

**Comment by abosii-harness** [comment]
trigger checkstylecodeformat, commonchecks, uts

**Comment by abosii-harness** [comment]
trigger messagecheck

**Pull request merged**